lib/inferno/dsl/fhir_resource_validation.rb

```diff
@@ -231,6 +233,9 @@ def issue_message(issue, resource)
 
         # @private
         def wrap_resource_for_hl7_wrapper(resource, profile_url)
+          validator_session_id = \
+            Inferno::Repositories::ValidatorSessions.new.validator_session(test_suite_id,
```

Comment:
What is the circumstance under which this would return `nil` and rely on the instance variable? If it doesn't return `nil`, wouldn't the instance variable need to be updated?

Reply:
I could imagine a situation where the startup invoke-session process fails for some reason but we still want to continue and try validating again just in case. But if that happens, this call would return `nil` but the instance variable would also be nil. I don't think it's possible for the instance variable to be set but not get a result from the DB.

Also, good point about updating the instance variable. We may not keep all sessions forever, so if a session expires, the next validation call will return with a new session id. If the new session id != the old one then it should be persisted.

Reply:
I figured this was in case the DB fails temporarily for some reason. Updating the instance variable makes sense here. It also gets set in `operation_outcome_from_hl7_wrapped_response()`, but that is after the response is received. In theory these would be sync'd up.